src/Microsoft.Dnx.Compilation.CSharp/RoslynProjectReference.cs

```diff
@@ -68,45 +69,95 @@ public IList<ISourceReference> GetSources()
 
-                Logger.TraceInformation("[{0}]: Emitting assembly for {1}", GetType().Name, Name);
+            var mainCompilation = afterCompileContext.Compilation;
+            afterCompileContext.Compilation = CSharpCompilation.Create(
+                assemblyName.Name,
+                options: mainCompilation.Options);
 
-                var sw = Stopwatch.StartNew();
+            Logger.TraceInformation("[{0}]: Emitting assembly for {1}", GetType().Name, Name);
 
-                EmitResult emitResult = null;
+            var sw = Stopwatch.StartNew();
 
-                if (_supportsPdbGeneration.Value)
-                {
-                    emitResult = CompilationContext.Compilation.Emit(assemblyStream, pdbStream: pdbStream, manifestResources: resources);
-                }
-                else
-                {
-                    Logger.TraceWarning("PDB generation is not supported on this platform");
-                    emitResult = CompilationContext.Compilation.Emit(assemblyStream, manifestResources: resources);
-                }
+            var emitResult = afterCompileContext.Compilation.Emit(
+                    afterCompileContext.AssemblyStream,
+                    manifestResources: resources);
 
-                sw.Stop();
+            sw.Stop();
 
-                Logger.TraceInformation("[{0}]: Emitted {1} in {2}ms", GetType().Name, Name, sw.ElapsedMilliseconds);
+            Logger.TraceInformation("[{0}]: Emitted {1} in {2}ms", GetType().Name, Name, sw.ElapsedMilliseconds);
 
-                var diagnostics = CompilationContext.Diagnostics.Concat(
-                    emitResult.Diagnostics);
+            afterCompileContext.Diagnostics = CompilationContext.Diagnostics
+                .Concat(emitResult.Diagnostics)
+                .ToList();
 
+            return emitResult;
+        }
+
+        public Assembly Load(AssemblyName assemblyName, IAssemblyLoadContext loadContext)
+        {
+            using (var pdbStream = new MemoryStream())
+            using (var assemblyStream = new MemoryStream())
+            {
                 var afterCompileContext = new AfterCompileContext
                 {
                     ProjectContext = CompilationContext.ProjectContext,
                     Compilation = CompilationContext.Compilation,
                     AssemblyStream = assemblyStream,
-                    SymbolStream = pdbStream,
-                    Diagnostics = new List<Diagnostic>(diagnostics)
+                    SymbolStream = pdbStream
                 };
 
+                EmitResult emitResult = null;
+
+                if (!string.IsNullOrEmpty(assemblyName.CultureName))
+                {
+                    var resourcesForCulture = ResourcesForCulture.GetResourcesForCulture(assemblyName.CultureName ?? string.Empty, CompilationContext.Resources);
+                    if (resourcesForCulture == null)
```

Comment:
This happens when user calls `ResourceManager` for a culture that does not have a corresponding `resx` file. `ReadResourcesFromProjectNugetPackageAndClassLibrary` test covers this case.